src/Features/LanguageServer/Protocol/CustomProtocol/RoslynCompletionItem.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Runtime.Serialization;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.CustomProtocol
+{
+    /// <summary>
+    /// A completion item to merge <see cref="VSCompletionItem"/> and custom tags for roslyn completion item.
+    /// This is an implementation detail of the server that is passed to the clients
+    /// and returned back without the clients parsing it, so no need to make it public.
+    /// </summary>
+    [DataContract]
+    internal class RoslynCompletionItem : CompletionItem
```

Comment:
in the meeting with Sri, I think we decide not to support this at all in Roslyn?

Reply:
I was actually looking into it after the meeting.  Right now they return a vscompletionitem if it's vs16 and the client capabilities has a visual studio lsp capability, otherwise they return the Roslyn completion item.  If we only plan to support 16 then we can remove it (and the client capabilities check).